File: android/app/src/main/java/org/pathcheck/covidsafepaths/bridge/RealmManager.kt
package org.pathcheck.covidsafepaths.bridge

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableArray
import org.pathcheck.covidsafepaths.storage.Location.Companion.SOURCE_GOOGLE
import org.pathcheck.covidsafepaths.storage.Location.Companion.SOURCE_MIGRATION
import org.pathcheck.covidsafepaths.storage.RealmWrapper

class RealmManager(reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

  override fun getName(): String {
    return "RealmManager"
  }

  @ReactMethod
  fun getLocations(promise: Promise) {
    RealmWrapper.getLocations(promise)
  }

  @ReactMethod
  fun importGoogleLocations(locations: ReadableArray, promise: Promise) {
    RealmWrapper.importLocations(locations, SOURCE_GOOGLE, promise)
  }

  @ReactMethod
  fun migrateExistingLocations(locations: ReadableArray, promise: Promise) {
    RealmWrapper.importLocations(locations, SOURCE_MIGRATION, promise)
  }
}

File: ios/RealmManager.swift
//
//  RealmManager.swift
//  COVIDSafePaths
//
//  Created by Tyler Roach on 4/29/20.
//  Copyright © 2020 Path Check Inc. All rights reserved.
//

import Foundation

@objc(RealmManager)
class RealmManager: NSObject {
  
  @objc
  func getLocations(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    RealmWrapper.shared.getLocations(resolve: resolve, reject: reject)
  }
  
  @objc
  func importGoogleLocations(_ locations: NSArray, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    RealmWrapper.shared.importLocations(locations: locations, source: Location.SOURCE_GOOGLE, resolve: resolve, reject: reject)
  }
  
  @objc
  func migrateExistingLocations(_ locations: NSArray, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    RealmWrapper.shared.importLocations(locations: locations, source: Location.SOURCE_MIGRATION, resolve: resolve, reject: reject)
  }

}
